core/src/main/java/org/keycloak/representations/idm/UserRepresentation.java

```diff
@@ -287,4 +288,34 @@ public void setDisableableCredentialTypes(Set<String> disableableCredentialTypes
     public void setAccess(Map<String, Boolean> access) {
         this.access = access;
     }
+
+    public Map<String, List<String>> toAttributes() {
+        Map<String, List<String>> attrs = new HashMap<>();
+
+        if (getAttributes() != null) attrs.putAll(getAttributes());
+
+        if (getUsername() != null)
+            attrs.put("username", Collections.singletonList(getUsername()));
+        else
+            attrs.remove("username");
+
+        if (getEmail() != null)
+            attrs.put("email", Collections.singletonList(getEmail()));
+        else
+            attrs.remove("email");
+
+        if (getUsername() != null)
```

Comment:
This looks duplicate, username is filled into attrs at line 297 already

Reply:
Going to remove it. Please, see my previous comment. We could remove this and have it in a single pace.